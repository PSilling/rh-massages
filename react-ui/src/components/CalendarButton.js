// react imports
import React, { Component } from 'react';

/**
 * Custom button component that handles event addition to Google Calendar.
 */
class CalendarButton extends Component {

  render() {
    return(
      <span>
        <button disabled={this.props.disabled} style={{ 'color': '#000' }}  type="button"
          className="btn btn-link" onClick={this.props.onAdd}>
          <span className="glyphicon glyphicon-calendar"></span>
        </button>
      </span>
    );
  }
}

export default CalendarButton
