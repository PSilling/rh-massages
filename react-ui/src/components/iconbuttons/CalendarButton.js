// react imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// util imports
import _t from '../../util/Translations';

/**
 * Icon only button that handles event addition to Google Calendar.
 */
class CalendarButton extends Component {

  render() {
    return (
      <span>
        {this.props.disabled ?
          <button disabled style={{ 'color': '#000' }}  type="button"
            className="btn btn-link" title={ _t.translate("Add to Google Calendar") }>
            <span className="glyphicon glyphicon-calendar"></span>
          </button> :
          <a href={this.props.link} target="_blank" tabIndex="-1">
            <button style={{ 'color': '#000' }}  type="button"
              className="btn btn-link" title={ _t.translate("Add to Google Calendar") }>
              <span className="glyphicon glyphicon-calendar"></span>
            </button>
          </a>
        }
      </span>
    );
  }
}

CalendarButton.propTypes = {
  /** calendar link to redirect to */
  link: PropTypes.string,
  /** whether the button should be disabled */
  disabled: PropTypes.bool
};

CalendarButton.defaultProps = {
  link: "",
  disabled: false
};

export default CalendarButton
