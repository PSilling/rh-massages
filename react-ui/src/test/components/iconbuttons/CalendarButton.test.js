// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import CalendarButton from '../../../components/iconbuttons/CalendarButton';

test('renders content with correct props', () => {
  const testRenderer = TestRenderer.create(
          <CalendarButton link="test" />
        ),
        testInstance = testRenderer.root;

  let link = testInstance.findByType('a'),
      icon = testInstance.findByProps({ className: "glyphicon glyphicon-calendar" }),
      treeJSON = testRenderer.toJSON();

  expect(link.props.href).toBe("test");
  expect(treeJSON).toMatchSnapshot();
});

test('changes button content when disabled', () => {
  const testRenderer = TestRenderer.create(
          <CalendarButton link="test" disabled />
        ),
        testInstance = testRenderer.root;

  let links = testInstance.findAllByType('a'),
      icon = testInstance.findByProps({ className: "glyphicon glyphicon-calendar" }),
      treeJSON = testRenderer.toJSON();

  expect(links).toHaveLength(0);
  expect(treeJSON).toMatchSnapshot();
});
