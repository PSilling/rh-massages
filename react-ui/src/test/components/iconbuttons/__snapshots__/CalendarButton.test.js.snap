// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes button content when disabled 1`] = `
<span>
  <button
    className="btn btn-link"
    disabled={true}
    style={
      Object {
        "color": "#000",
      }
    }
    title="Přidat do Google kalendáře"
    type="button"
  >
    <span
      className="glyphicon glyphicon-calendar"
    />
  </button>
</span>
`;

exports[`renders content with correct props 1`] = `
<span>
  <a
    href="test"
    tabIndex="-1"
    target="_blank"
  >
    <button
      className="btn btn-link"
      style={
        Object {
          "color": "#000",
        }
      }
      title="Přidat do Google kalendáře"
      type="button"
    >
      <span
        className="glyphicon glyphicon-calendar"
      />
    </button>
  </a>
</span>
`;
